### PR TITLE
refactor the CI + testing setup

### PR DESCRIPTION
**Pull request by devoncarew**
refactor the CI + testing setup:
- consolidate into fewer testing configs; one main config - `test` - and additional ones for the flavor and architecture params
- use the example/ project in this repo for more of the testing (instead of stagehand)
- fix https://github.com/dart-lang/setup-dart/issues/73

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -5,7 +5,11 @@ branding:
   color: blue
 inputs:
   sdk:
-    description: "The channel, or a specific version from a channel to install ('stable', 'beta', 'dev', '2.7.2', '2.12.0-1.4.beta'). Using one of the three channels instead of a version will give you the latest version published to that channel."
+    description:
```
Comment (kevmoo) [inline action.yml:8]:
```suggestion
    description: >-
```

per https://yaml-multiline.info/

Comment (devoncarew) [reply action.yml:8]: Thanks!

File: .github/workflows/dart.yml [hunk 2]
```diff
@@ -9,42 +9,59 @@ on:
...
+    defaults:
+      run:
+        working-directory: example
     strategy:
-      fail-fast: false
       matrix:
         os: [ubuntu-latest, macos-latest, windows-latest]
-        # Test latest stable, beta, dev channels + last stable release
-        # prior to the introduction of the unified `dart` developer tool.
-        sdk: [stable, beta, dev, 2.9.3, 2.12.0-29.10.beta]
```
Comment (athomas) [inline .github/workflows/dart.yml:20]: The specific beta case was testing this branch:
https://github.com/dart-lang/setup-dart/blob/main/setup.sh#L27

If we have no case were we test "detect channel from version anymore" we lose coverage.

**Review by athomas — APPROVED** [review]

File: .github/workflows/dart.yml (hunk 2 above)
Comment (devoncarew) [reply .github/workflows/dart.yml:20]: Gotcha - I'll update this PR to test inferring the channel from the sdk param.

**Pull request merged**